### PR TITLE
Embed "How do I ... add tests?" video in developer docs

### PR DESCRIPTION
**Pull request by nrichers**
## Internal Notes for Reviewers

This PR adds a new "How do I ... add tests?" FAQ-style video to our developer content. This is a single, short FAQ-style video that answers a specific question. This one covers:

- Exploring tests
- Using ValidMind to filter for tests
- Adding tests to your model documentation

[**LIVE PREVIEW**](https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6095/how-to-video-for-run-tests/developer/model-testing/testing-overview.html#explore-tests-and-test-suites)

image

Note this video embed does not include lead-in text. Some had been added to https://github.com/validmind/documentation/pull/446 but I removed it again for the sake of simplicity. 

(Related, I am not clear on what alt tests we might need to provide for videos. YouTube does generate captions but it's something to look into.)

## External Release Notes

We added a short FAQ-style video to our developer documentation that shows you how to find tests ValidMind provides and add them to your own model documentation. [Watch it ...](../../developer/model-testing/testing-overview.html#explore-tests-and-test-suites)

### CODE REVIEW

File: site/developer/model-testing/testing-overview.qmd [hunk 1]
```diff
@@ -78,6 +78,8 @@ Next, find available tests and test suites using the {{< var vm.developer >}} or
 :::{#tests-explore}
 :::
 
+{{< video https://youtu.be/5J7wysDIXgI width="560" height="315" >}}
```
Comment (nrichers) [inline site/developer/model-testing/testing-overview.qmd:81]: @validbeck related to your visual tweak on the other video PR ... do we need to add something to our CSS for the drop shadow and rounded corners when we use the `{{< video >}}` directive?

Comment (validbeck) [reply site/developer/model-testing/testing-overview.qmd:81]: Yes, please copy and paste the code from the other PR and just change the URL for that here. I actually based that styling off of the video styling you did in the releases, it's just cleaned up a little.

**Review by validbeck — CHANGES_REQUESTED** [review]
Again, I'm not a fan of NOT having lead-in text to videos. It should be clear what the content of the videos are and what value they bring that supplements or differs from the written docs.

File: site/developer/model-testing/testing-overview.qmd (hunk 1 above)
Comment (nrichers) [reply site/developer/model-testing/testing-overview.qmd:81]: The other PR uses an iframe, but I'll see what I can do.

Comment (validbeck) [reply site/developer/model-testing/testing-overview.qmd:81]: Yes, you can just use the iframe here as well — I ended up replacing the quickstart video with that version of the code, for example: https://docs.validmind.ai/get-started/developer/try-with-jupyterhub.html

**Comment by nrichers** [comment]
> Again, I'm not a fan of NOT having lead-in text to videos. It should be clear what the content of the videos are and what value they bring that supplements or differs from the written docs.

Before I go ahead and add lead-in text to address your concerns, let me explain why I think it isn't necessary here:

- The videos always include a text title that clearly state what they explain — the context is generally unambiguous.
- We don't use lead-in text for some animated GIFs either, which is the closest analogue to videos we already use.
- Many other documentation sites don't include lead-in text for their videos:
   - [Stripe example](https://docs.stripe.com/no-code/get-started#sell-online)
   - [DataDog example](https://docs.datadoghq.com/watchdog/)
   - [Grafana](https://grafana.com/docs/grafana-cloud/get-started/)

Finally, lead-in text had a storied life before becoming passé around 2010. At one point, our editors required it for many things at IBM, for instance, you couldn't have a heading followed immediately by a list without lead-in text. Then, one year, a revolution happened, and writing minimalism replaced many of those guidelines almost overnight. If the context was clear, the new guidance was to omit lead-in text. (And then IBM decided it didn't need editors, either, but that's a separate story.)

That said, if I still haven't convinced you, I'll add some lead-in text. 🙂

File: site/developer/model-testing/testing-overview.qmd (hunk 1 above)
Comment (nrichers) [reply site/developer/model-testing/testing-overview.qmd]: Roger, I switched to an iframe with the same CSS tweaks for the video embed.

**Comment by validbeck** [comment]
Thank you for the explanation as to your logic!

I think the video titles are more vague to me than they are to you, so that's where we disagree. I don't think it "makes or breaks" things, and if the interest is lightness, then that's fine with me — just letting you know that personally as a user, I'm not inclined to click on videos unless there's some explanation as to why I should spend my time watching it.

**Comment by validbeck** [comment]
Ah, I see what it is my brain is flagging.

Screenshot 2024-09-26 at 12 28 33 PM

The `How do I ... add tests?` with the ellipsis is what is throwing me off here. The ellipsis makes it feel like the video title is truncated, rather than what it's actually supposed to be.

Can we at least retitle the video filename on YouTube to just "How do I add tests?"

**Comment by nrichers** [comment]
> Ah, I see what it is my brain is flagging.
> Can we at least retitle the video filename on YouTube to just "How do I add tests?"

Yes, and that's very good feedback! Maybe a good middle ground would be to require meaningful video titles that are easy to parse. 

I can also do one better and expand the video title a bit: 

image

Would this work for you for now?

**Review by validbeck — APPROVED** [review]

**Comment by nrichers** [comment]
@mehdi0501 I'm going to merge this, but if you have feedback, please open a story or let me know what the feedback is and I will open a story to make sure it's addressed.

**Pull request merged**